### PR TITLE
Add bind rule to the flutter App

### PR DESCRIPTION
**Pull request by ErebusZ**
This PR adds a bug rule for the method Bind, this should trigger an open port on device finding.

### CODE REVIEW

**Comment by ohachimOs** [comment]
`bind` fails so we catch the error to avoid crash? can we maybe attempt to bind to another port? or just wait a bit a try again, sometimes there is a bit of lag with the `TIME_WAIT` state, if the port was recently bind to something (it could be the same app that had just closed), is it not worth it to try in this use case?
Also can you add a description

**Comment by ErebusZ** [comment]
> TIME_WAIT

That is a little bit behind the point of the app, this code is meant for testing the platform detection ability. invoking bind with vulnerable args is enough to do that.

**Comment by 3asm** [comment]
@ErebusZ please add a description.

File: ostorlab_insecure_flutter_app/lib/bugs/bind.dart [hunk 1]
```diff
@@ -0,0 +1,31 @@
+import 'dart:io';
+import 'package:ostorlab_insecure_flutter_app/bug_rule.dart';
+
+/// A [BugRule] that triggers the use of bind.
+class OpenPortOnDevice extends BugRule {
```
Comment (3asm) [inline ostorlab_insecure_flutter_app/lib/bugs/bind.dart]: I would rather have it name BindRule, I would know what it does.

**Review by 3asm — CHANGES_REQUESTED** [review]

**Comment by ErebusZ** [comment]
> @ErebusZ please add a description.

Added.

File: ostorlab_insecure_flutter_app/lib/bugs/bind.dart [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+  static const String _tag = 'BindRule';
+
+  /// Returns a description of this [BugRule] implementation.
+  @override
+  String get description => 'Bind local host to a port';
+
+  /// Trigger the [BugRule] by starting an HTTP server that listens for connections and binds to a port.
+  @override
+  Future<void> run(String input) async {
+    var port = (input != "") ? int.parse(input) : 450;
```
Comment (3asm) [inline ostorlab_insecure_flutter_app/lib/bugs/bind.dart]:
```suggestion
    var port = (input != "") ? int.parse(input) : 4450;
```

**Review by 3asm — APPROVED** [review]

**Pull request merged**